### PR TITLE
Improve syntax highlighting.

### PR DESCRIPTION
**Pull request by yuchenshi**
... What do you mean by "it's Friday night"?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,16 +33,16 @@
...
-    "@types/codemirror": "0.0.106",
+    "@types/codemirror": "^5.60.1",
     "@types/jest": "^26.0.0",
     "@types/lodash.debounce": "^4.0.6",
     "@types/node": "^13.9.3",
     "@types/react": "^17.0.0",
     "@types/react-dom": "^17.0.0",
     "@types/react-redux": "^7.1.7",
     "@types/recompose": "^0.30.7",
-    "@uiw/react-codemirror": "^3.0.2",
```
Comment (yuchenshi) [inline package.json:44]: Our use case is a simple one and don't need a wrapper. It only makes logic harder to understand.

File: src/components/Firestore/Requests/RequestDetails/CodeViewer/mode.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+const FIREBASE_RULES_MISC_REGEX = /(?:[.,:[\]{}()])/;
+const FIREBASE_RULES_STRING_REGEX = /(?:auth|create|delete|get|headers|method|params|path|read|request|resource|time|uid|update|write|list)\b/;
+
+export const MODE_FIREBASE_RULES = 'firebaseRules';
+
+export function defineFirebaseRulesMode() {
+  CodeMirror.defineSimpleMode(MODE_FIREBASE_RULES, {
+    start: [
+      { regex: FIREBASE_RULES_COMMENT_REGEX, token: 'comment' },
+      { regex: /\/\*/, token: 'comment', next: 'comment' },
```
Comment (yuchenshi) [inline src/components/Firestore/Requests/RequestDetails/CodeViewer/mode.ts:39]: Added support for multi-line comments. Unfortunately there are edge cases like `match /foo/bar/*` (note the star) which I hope nobody run into. Feel free to copy this back to Console.

File: src/components/Firestore/Requests/RequestDetails/CodeViewer/index.scss [hunk 3]
```diff
@@ -81,3 +85,140 @@
...
+  background-color: #ebebeb;
+}
+
+.cm-s-devsite-light .CodeMirror-activeline-background {
+  background-color: rgba(
+    3,
+    155,
+    229,
+    0.1
+  ); /* GMP_COLOR_PRIMARY_DEFAULT @ 10% */
```
Comment (sam-gc) [inline src/components/Firestore/Requests/RequestDetails/CodeViewer/index.scss:126]: Let's get rid of this comment
```suggestion
  );
```

File: src/components/Firestore/Requests/RequestDetails/CodeViewer/index.scss [hunk 4]
```diff
@@ -81,3 +85,140 @@
...
+}
+
+.cm-s-devsite-light .CodeMirror-foldmarker {
+  color: #bf360c;
+  text-shadow: #bf360c 1px 1px 2px, #bf360c -1px -1px 2px, #bf360c 1px -1px 2px,
+    #bf360c -1px 1px 2px;
+}
+
+.cm-s-devsite-light .cm-trailingspace {
+  border-bottom: 1px dashed red; /* TODO(tstirrat): needs better style */
```
Comment (sam-gc) [inline src/components/Firestore/Requests/RequestDetails/CodeViewer/index.scss:142]: And this one
```suggestion
  border-bottom: 1px dashed red;
```

File: src/components/Firestore/Requests/RequestDetails/CodeViewer/index.tsx [hunk 5]
```diff
@@ -15,18 +15,21 @@
...
+import React, { useEffect, useRef } from 'react';
 
 import { textBlackTertiary } from '../../../../../colors';
 import { errorTheme, successTheme } from '../../../../../themes';
 import { OutcomeInfo, RulesOutcome } from '../../rules_evaluation_result_model';
 import { ICON_SELECTOR } from '../../utils';
 import CodeViewerAdminRequest from './AdminRequest';
+import { MODE_FIREBASE_RULES, defineFirebaseRulesMode } from './mode';
+
+// import CodeMirror from '@uiw/react-codemirror';
```
Comment (sam-gc) [inline src/components/Firestore/Requests/RequestDetails/CodeViewer/index.tsx:32]: Unused line
```suggestion

```

File: src/components/Firestore/Requests/RequestDetails/CodeViewer/index.scss [hunk 6]
```diff
@@ -81,3 +85,140 @@
     width: 61px !important;
   }
 }
+
+/* CodeMirror theme, based on http://go/firebase-console-codemirror-theme */
```
Comment (sam-gc) [inline src/components/Firestore/Requests/RequestDetails/CodeViewer/index.scss:89]: Was the go link intentional?

**Review by sam-gc — APPROVED** [review]

File: src/components/Firestore/Requests/RequestDetails/CodeViewer/index.scss (hunk 6 above)
Comment (yuchenshi) [reply src/components/Firestore/Requests/RequestDetails/CodeViewer/index.scss:89]: Yeah, I created the go-link just for this comment

File: src/components/Firestore/Requests/RequestDetails/CodeViewer/index.tsx (hunk 5 above)
Comment (yuchenshi) [reply src/components/Firestore/Requests/RequestDetails/CodeViewer/index.tsx]: Removed

**Pull request merged**